processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java

```diff
@@ -55,7 +55,7 @@ public RegisteredLookupExtractionFn(
     this.replaceMissingValueWith = replaceMissingValueWith;
     this.retainMissingValue = retainMissingValue;
     this.injective = injective;
-    this.optimize = optimize;
+    this.optimize = optimize == null ? true : optimize;
```

Comment:
I take no stand on what the default should be, other than that they should behave the same way. Happy to change them both to default false.